core/templates/pages/email-dashboard-pages/email-dashboard-page.component.ts

```diff
@@ -21,34 +21,39 @@ require('base-components/base-content.directive.ts');
 require('services/user.service.ts');
 
 require('./email-dashboard-data.service');
+require(
+  'components/forms/schema-based-editors/schema-based-editor.directive.ts');
 
 angular.module('oppia').component('emailDashboardPage', {
   template: require('./email-dashboard-page.component.html'),
   controller: [
     '$rootScope', 'EmailDashboardDataService', 'LoaderService',
-    'UserService', function(
+    'UserService', 'EMAIL_DASHBOARD_PREDICATE_DEFINITION', function(
         $rootScope, EmailDashboardDataService, LoaderService,
-        UserService) {
+        UserService, EMAIL_DASHBOARD_PREDICATE_DEFINITION) {
       var ctrl = this;
+      ctrl.notRequired = true;
       ctrl.resetForm = function() {
-        ctrl.hasNotLoggedInForNDays = null;
-        ctrl.inactiveInLastNDays = null;
-        ctrl.createdAtLeastNExps = null;
-        ctrl.createdFewerThanNExps = null;
-        ctrl.editedAtLeastNExps = null;
-        ctrl.editedFewerThanNExps = null;
+        ctrl.data = [];
+        EMAIL_DASHBOARD_PREDICATE_DEFINITION.forEach(predicate => {
+          ctrl.data.push({
+            attribute: predicate.backend_attr,
+            value: predicate.default_value,
+          });
+        });
+      };
+
+      ctrl.areAllInputsEmpty = function() {
+        for (var i = 0; i < ctrl.data.length; i++) {
+          if (ctrl.data[i].value !== null) {
+            return false;
+          }
+        }
+        return true;
```

Comment:
You can do `ctrl.data.every(datum => { return datum.value !== null })` (with better formatting) instead.

Reply:
Done.